src/cmd/cli/command/commands.go

```diff
@@ -290,34 +286,9 @@ var RootCmd = &cobra.Command{
 			}
 		}
 
-		var projectName string
-		if _, ok := cmd.Annotations[projectNeeded]; ok {
-			filePath, _ := cmd.InheritedFlags().GetString("file")
-			projectName = os.Getenv("COMPOSE_PROJECT_NAME") // overrides the project name, except in the playground env
-			if projectName != "" {
-				project, err = cli.LoadComposeWithProjectName(filePath, projectName)
-			} else {
-				tenantID := cli.GetTenantID(cluster)
-				projectName = string(tenantID)
-				if provider == cliClient.ProviderDefang {
-					project, err = cli.LoadComposeWithProjectName(filePath, projectName) // playground env
-				} else {
-					project, err = cli.LoadCompose(filePath, tenantID) // fallback to tenant ID
-				}
-			}
-
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					term.Debug(" - Could not find docker compose file: ", err)
-				} else {
-					term.Warn(" - Error loading docker compose file: ", err)
-				}
-			} else {
-				projectName = project.Name
-			}
-		}
-
-		client = cli.NewClient(cluster, projectName, provider)
+		project.ComposeFilePath, _ = cmd.Flags().GetString("file")
+		project.TenantID = cli.GetTenantID(cluster)
```

Comment:
we now have `WhoAmI` so we could probably remove this hack `cli.GetTenantID`

Reply:
After restructuring, the TenantID is no longer needed here, but I think the right place for us to get the tenantID from fabric would be on connect.go @ L68, see my comments below.

Reply:
I just added the tenant ID reloading, let me know what do you think